lsstswBuild.sh

```diff
@@ -18,15 +18,92 @@ BUILD_DOCS="yes"
 RUN_DEMO="yes"
 PRODUCT=""
 NO_FETCH=0
+PRINT_FAIL=0
+COLORIZE=0
 
-# Buildbot remotely invokes scripts with a stripped down environment.  
+# Buildbot remotely invokes scripts with a stripped down environment.
 umask 002
 
+sbar=$(printf %78s |tr " " "-")
+tbar=$(printf %78s |tr " " "~")
+
+set_color() {
+    if [ $COLORIZE -eq 1 ]; then
+        echo -ne "$@"
+    fi
+}
+
+no_color() {
+    if [ $COLORIZE -eq 1 ]; then
+        echo -ne $NO_COLOR
+    fi
+}
+
+print_success() {
+    set_color $LIGHT_GREEN
+    echo -e "$@"
+    no_color
+}
+
+print_info() {
+    set_color $YELLOW
+    echo -e "$@"
+    no_color
+}
+
+# XXX this script is very inconsistent about what is sent to stdout vs stderr
 print_error() {
-    >&2 echo $@
+    set_color $LIGHT_RED
+    >&2 echo -e "$@"
+    no_color
 }
 
-options=(getopt --long build_number:,branch:,product:,skip_docs,skip_demo,no-fetch -- "$@")
+start_section() {
+    print_info "### $@"
+    print_info $tbar
+}
+
+end_section() {
+    print_info $sbar
+    echo -ne "\n"
+}
+
+print_build_failure() {
+    for pkg in "${PACKAGES[@]}"; do
+        build_dir=${LSSTSW_BUILD_DIR}/${pkg}
+        build_log=${build_dir}/_build.log
+        test_dir=${build_dir}/tests/.tests
+
+        # check to see if build log exists
+        if [ ! -e $build_log ]; then
+            # no log means that package was not [attempted] to be be built
+            # on this run
+            continue
+        fi
+
+        # check to see if package had any failed tests
+        shopt -s nullglob
+        failed_tests=(${test_dir}/*.failed)
+
+        if [ ${#failed_tests[@]} -ne 0 ]; then
+            # only print the build log if there were test failures
```

Comment:
Listing the contents of tens of build log files seems scary when the build only failed in the last. Are you completely against sorting them by timestamp and picking the most recently changed build.log file?